File: pinyougou_shop_web/src/main/webapp/js/service/goodsService.js
//服务层
app.service('goodsService',function($http){
	this.init = ()=>{
		return $http.get("/goods/findAll.do");
	}
	this.addInfo = (entity)=>{
		return $http.post("/goods/addInfo.do",entity);
	}
	this.getLevel = (id)=>{
		return $http.get("/goods/getLevel.do?id="+id);
	}
	this.getTemplateInfo =(templateId)=>{
		return $http.get("/goods/getTemplateInfo.do?id="+templateId);
	}
	this.getspecs = (ids)=>{
		return $http.get("/goods/getspecs.do?ids="+ids);
	}
	this.getCategory = () => {
		return $http.get("/goods/addCategoryList.do");
	}
	this.getLevelAllInfo =()=>{
		return $http.get("/goods/getLevelAllInfo.do");
	}
	this.uploadFile = ()=>{
		var data = new FormData();
		data.append("file",file.files[0]);
		return $http({
			url:'../upload.do',
			data:data,
			method:'post',
			headers:{'Content-Type':undefined},
			transformRequest:angular.identity
		})
	}
});

File: pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/GoodsService.java
package com.pinyougou.sellergoods.service;


import com.pinyougou.group.Goods;
import com.pinyougou.pojo.Result;
import com.pinyougou.pojo.TbItem;
import com.pinyougou.pojo.TbTypeTemplate;

import java.util.List;
import java.util.Map;

public interface GoodsService {


	List<Goods> getAll();

    List<TbItem> getAllItems();

    Result addInfo(Goods goods);

    List getLevel(long id);

    TbTypeTemplate getTemplateInfo(long id);

    List getspecs(long[] ids);

    List addCategoryList();

    List getLevelAllInfo();
}
